src/commands/dev/index.js

```diff
@@ -283,8 +285,9 @@ async function serveRedirect(req, res, proxy, match, options) {
     return render404(options.publicFolder)
   }
 
-  if (match.force || (!(await isStatic(reqUrl.pathname, options.publicFolder) || options.framework) && match.status !== 404)) {
```

Comment:
There are two main changes here. `|| options.framework` should've been `&& options.framework` which was a mistake and we've added tests to catch it in the future. And `match.status !== 404` shouldn't really be here. I'm not sure why exactly it is here because we want to overwrite `status` for `404` rewrites but we're added tests to make sure the intended functionality is preserved.